Buildings/Controls/OBC/Utilities/PIDWithAutotuning/PIDWithAutotuningAmigoFOTD.mo

```diff
@@ -0,0 +1,232 @@
+  parameter Real Td_start(
+    final quantity="Time",
+    final unit="s",
+    min=100*Buildings.Controls.OBC.CDL.Constants.eps)=0.1
+    "Start value of the time constant of derivative block"
+    annotation (Dialog(group="Control gains",enable=controllerType == CDL.Types.SimpleController.PD or controllerType == CDL.Types.SimpleController.PID));
+  parameter Real yHig(min=1E-6) = 1
+    "Higher value for the output";
+  parameter Real yLow(min=1E-6) = 0.5
+    "Lower value for the output";
+  parameter Real deaBan(min=1E-6) = 0.5
+    "Deadband for holding the output value";
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput u_s
+    "Connector for setpoint input signal"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}}),iconTransformation(extent={{-140,-20},{-100,20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput u_m
+    "Connector for measurement input signal"
+    annotation (Placement(transformation(origin={0,-120},extent={{20,-20},{-20,20}},rotation=270),iconTransformation(extent={{20,-20},{-20,20}},rotation=270,origin={0,-120})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput trigger
+    "Resets the controller output when trigger becomes true"
+    annotation (Placement(transformation(extent={{-20,-20},{20,20}},rotation=90,origin={-60,-120}), iconTransformation(extent={{-20,-20},{20,20}},rotation=90,origin={-60,-120})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput y
+    "Connector for actuator output signal"
+    annotation (Placement(transformation(extent={{100,-20},{140,20}}),iconTransformation(extent={{100,-20},{140,20}})));
+  Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.Controller relay(
+    yHig=yHig,
+    yLow=yLow,
+    deaBan=deaBan)
+    "A relay controller"
+    annotation (Placement(transformation(extent={{22,20},{42,40}})));
+  Buildings.Controls.OBC.Utilities.PIDWithInputGains pid(
+     controllerType=controllerType) "A pid controller"
+    annotation (Placement(transformation(extent={{22,-40},{42,-20}})));
+  Buildings.Controls.OBC.CDL.Continuous.Switch swi
+    "Switch between a PID controller and a relay controller"
+    annotation (Placement(transformation(extent={{62,10},{82,-10}})));
+  Buildings.Controls.OBC.CDL.Discrete.TriggeredSampler samk(y_start=k_start)
+    "Recording the control gain"
+    annotation (Placement(transformation(extent={{-40,-10},{-20,-30}})));
+  Buildings.Controls.OBC.CDL.Discrete.TriggeredSampler samTi(y_start=Ti_start) if with_I
+    "Recording the integral time"
+    annotation (Placement(transformation(extent={{-80,-38},{-60,-58}})));
+  Buildings.Controls.OBC.CDL.Discrete.TriggeredSampler samTd(y_start=Td_start) if with_D
+    "Recording the derivative time"
+    annotation (Placement(transformation(extent={{-40,-60},{-20,-80}})));
+  Buildings.Controls.OBC.Utilities.PIDWithAutotuning.SystemIdentification.FirstOrderTimedelayed.ControlProcessModel
```

Comment:
the class name should be "ControlProcess", not "ControlProcessModel". It's redundant to add "Model" as the end.

Reply:
This is a little tricky. This class models the "reduced order model" for the control process not the control process itself.